Makefile.defs

```diff
@@ -132,7 +132,7 @@ ifneq ($(LOCKDEBUG),)
     GO_TEST_FLAGS += -tags lockdebug
 endif
 
-GO_BUILD_FLAGS += -ldflags '$(GO_BUILD_LDFLAGS) $(EXTRA_GO_BUILD_LDFLAGS)' $(EXTRA_GO_BUILD_FLAGS)
+GO_BUILD_FLAGS += -tags osusergo -ldflags '$(GO_BUILD_LDFLAGS) $(EXTRA_GO_BUILD_LDFLAGS)' $(EXTRA_GO_BUILD_FLAGS)
```

Comment:
Let's add `-tags osusergo` to `$GO_TEST_FLAGS` as well so were guaranteed to be using the non-cgo implementation in the tests as well.

Reply:
Of course, my mistake.

Reply:
No worries. Might make sense to introduce a separate `Makefile` variable for Go tags (e.g. `lockdebug` above) at some point which is then appended to both build and test flags. But that is of course outside the scope of this PR :smiley:

Reply:
Adding to `GO_TEST_FLAGS` also adds to `GO_CLEAN` which is probably not an issue but weird looking:$

```
go clean -mod=vendor -tags osusergo
```

Reply:
I think this is a but in the definition of `GO_CLEAN`:

Currently it says:
```make
GO_CLEAN = $(GO) clean $(GO_TEST_FLAGS)
```
but I think it should be:
```make
GO_CLEAN = $(GO) clean $(GO_CLEAN_FLAGS)
```